overlord/ifacestate/ifacestate.go

```diff
@@ -39,14 +39,28 @@ var noConflictOnConnectTasks = func(task *state.Task) bool {
 	return task.Kind() != "connect" && task.Kind() != "disconnect"
 }
 
+// AutoConnect returns a set of tasks for connection an interface as part of snap installation
+// and auto-connect handling.
+func AutoConnect(st *state.State, plugSnap, plugName, slotSnap, slotName string) (*state.TaskSet, error) {
+	return connect(st, plugSnap, plugName, slotSnap, slotName, true)
+}
+
 // Connect returns a set of tasks for connecting an interface.
 //
 func Connect(st *state.State, plugSnap, plugName, slotSnap, slotName string) (*state.TaskSet, error) {
-	if err := snapstate.CheckChangeConflict(st, plugSnap, noConflictOnConnectTasks, nil); err != nil {
-		return nil, err
-	}
-	if err := snapstate.CheckChangeConflict(st, slotSnap, noConflictOnConnectTasks, nil); err != nil {
-		return nil, err
+	return connect(st, plugSnap, plugName, slotSnap, slotName, false)
+}
+
+func connect(st *state.State, plugSnap, plugName, slotSnap, slotName string, autoConnect bool) (*state.TaskSet, error) {
+	// Check conflicts only if it's not autoconnect. Autoconnect can only happen during snap install, so potential
+	// conficts should already be known and prevented beforehand.
```

Comment:
That doesn't seem entirely true. We may want to use the auto-connect task in other circumstances than just installation. The better explanation here is that this task was created as an expansion of an earlier auto-connect task, so any conflicts must necessarily have been sorted before.

This also makes me feel like the name of the new function above may not be right. I'd expect an AutoConnect method here to create an auto-connect task, but that's not the case.

Reply:
that's not true also because connect has two sides, the conflict check of install just takes care of one, we ought to check for conflicts and have logic similar to the PR mvo is working on for installing default-providers.

It's true OTOH that differently than there this creates single tasks, not a sequence on multiple tasks, for the affected snaps but we would need to be very careful with corner cases if don't simply use the conflict logic.